browser/src/control/Control.Infobar.js

```diff
@@ -3,40 +3,20 @@
  * L.Control.Infobar used for displaying non-annoying info messages
  */
 
-/* global vex $ */
+/* global */
 L.Control.Infobar = L.Control.extend({
 	onAdd: function (map) {
 		map.on('infobar', this._onInfobar, this);
+		this.map = map;
 	},
 
 	_onInfobar: function(e) {
 		if (!e.msg)
 			return;
 
-
-		var buttons = [];
-		var callback = function() {};
-		if (e.actionLabel && e.action) {
-			buttons.push($.extend({}, vex.dialog.buttons.YES, { text: e.actionLabel, className: 'button-primary'}));
-			callback = function (value) {
-				if (value === false) // close btn clicked
-					return;
-
-				if (e.action.startsWith('http')) { // We have a link
-					var win = window.open(e.action, '_blank');
-					win.focus();
-				}
-			};
-		}
-
-		vex.dialog.open({
-			message: e.msg,
-			className: 'vex-theme-bottom-right-corner',
-			showCloseButton: true,
-			buttons: buttons,
-			callback: callback
-		});
+		this.map.uiManager.showInfoBar('info-bar', e.msg, e.action, e.actionLabel);
```

Comment:
what about removing whole Infobar file and replacing with that one line? what do you think?

Reply:
I also thought that we could indeed :) Let me remove that, thanks :)